pandas/tests/indexes/multi/test_join.py

```diff
@@ -96,10 +96,19 @@ def test_join_multi_wrong_order():
     midx1 = pd.MultiIndex.from_product([[1, 2], [3, 4]], names=["a", "b"])
     midx2 = pd.MultiIndex.from_product([[1, 2], [3, 4]], names=["b", "a"])
 
-    join_idx, lidx, ridx = midx1.join(midx2, return_indexers=False)
+    join_idx, lidx, ridx = midx1.join(midx2, return_indexers=True)
 
     exp_ridx = np.array([-1, -1, -1, -1], dtype=np.intp)
 
     tm.assert_index_equal(midx1, join_idx)
     assert lidx is None
     tm.assert_numpy_array_equal(ridx, exp_ridx)
+
+
+def test_join_multi_return_indexers():
+    # GH 34074
+
+    midx1 = pd.MultiIndex.from_product([[1, 2], [3, 4], [5, 6]], names=["a", "b", "c"])
+    midx2 = pd.MultiIndex.from_product([[1, 2], [3, 4]], names=["a", "b"])
+
+    assert type(midx1.join(midx2, return_indexers=False)) != tuple
```

Comment:
Could this be made more explicit?

Just a suggestion:
```python
result = midx1.join(midx2, return_indexers=False)
expected = <construct expected result here>
tm.assert_index_equal(result, expected)
```

As an aside, see [PEP8](https://www.python.org/dev/peps/pep-0008/) for how it's advised to write object type comparisons